acme_diags/driver/qbo_driver.py

```diff
@@ -0,0 +1,222 @@
+    data_lat_lon_average = cdutil.averager(data_region, axis='xy')
+    # Average over vertical
+    x0 = np.nanmean(np.array(data_lat_lon_average(level=(levTop, levBot))), axis=1)
+    # x0 should now be 1D
+    return x0
+
+
+def ceil_log2(x):
+    """
+    Given a number, calculate the exponent for the next power of 2.
+
+    Example:
+        ceil_log2(16) = 4
+        ceil_log2(17) = 5
+    """
+    return np.ceil(np.log2(x)).astype('int')
+
+
+def get_psd_from_deseason(xraw, period_new):
+    x_deseasoned = deseason(xraw)
+
+    # Sampling frequency: assumes frequency of sampling = 1 month
+    sampling_frequency = 1
+    # Calculate the period as a function of frequency
+    period0 = 1 / sampling_frequency
+    L0 = len(xraw)
+    t0 = np.arange(0, L0) * period0
+    NFFT0 = 2 ** ceil_log2(L0)
+
+    # Apply fft on x_deseasoned with n = NFFT
+    x0 = scipy.fftpack.fft(x_deseasoned, n=NFFT0) / L0
+    # Frequency (cycles/month). Frequency will be increasing.
+    frequency0 = sampling_frequency * np.arange(0, (NFFT0 / 2 + 1)) / NFFT0
+    # Period (months/cycle). Calculate as a function of frequency. Thus, period will be decreasing.
+    period0 = 1 / frequency0
+
+    # Calculate amplitude as a function of frequency
+    amplitude0 = 2 * abs(x0[0:int(NFFT0 / 2 + 1)])
+    # Calculate power spectral density as a function of frequency
+    psd_x0 = amplitude0 ** 2 / L0
+    period_end0 = period0 * L0
+    # Total spectral power
+    Pxf0 = period_end0 * np.sum(psd_x0)
+    # Later, we will perform an interpolation using the period
+    # (interpolating values of amplitude0_flipped and psd_x0_flipped from period0_flipped to period_new).
+    # For that interpolation, we want the period to be increasing.
+    # Therefore, we will flip the following values:
+    period0_flipped = period0[::-1]
+    amplitude0_flipped = amplitude0[::-1]
+    psd_x0_flipped = psd_x0[::-1]
+
+    amplitude_new0 = np.interp(period_new, period0_flipped[:-1], amplitude0_flipped[:-1])
+    psd_x_new0 = np.interp(period_new, period0_flipped[:-1], psd_x0_flipped[:-1])
+    return psd_x_new0, amplitude_new0
+
+
+def run_diag(parameter):
+    variables = parameter.variables
+    # The region will always be 5S5N
+    region = '5S5N'
+    test_data = utils.dataset.Dataset(parameter, test=True)
+    ref_data = utils.dataset.Dataset(parameter, ref=True)
+    for variable in variables:
+        if parameter.print_statements:
+            print('Variable={}'.format(variable))
+        test_var = test_data.get_timeseries_variable(variable)
+        ref_var = ref_data.get_timeseries_variable(variable)
+        qbo_region = default_regions.regions_specs[region]['domain']
+
+        test_region = test_var(qbo_region)
+        ref_region = ref_var(qbo_region)
+
+        test = {}
+        ref = {}
+
+        # Diagnostic 1: average over longitude & latitude to produce time-height array of u field:
+        # https://en.wikipedia.org/wiki/Quasi-biennial_oscillation
```

Comment:
Suggest to use following as reference for QBO diags
Richter, J. H., Chen, C. C., Tang, Q., Xie, S., & Rasch, P. J. (2019). Improved Simulation of the QBO in E3SMv1. Journal of Advances in Modeling Earth Systems, 11(11), 3403-3418.
https://agupubs.onlinelibrary.wiley.com/doi/epdf/10.1029/2019MS001763